pkg/controller/deployment/sync.go

```diff
@@ -59,6 +59,14 @@ func (dc *DeploymentController) sync(d *extensions.Deployment, rsList []*extensi
 		return err
 	}
 
+	// Clean up the deployment when it's paused but not when it's proportionally scaled
+	// in the middle of an on-going rollout.
```

Comment:
Maybe you'd want to update the comment and mention the `rollbackTo == nil` logic